pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -212,7 +211,7 @@ public Map<String, String> createTask(String taskType, String tableName, @Nullab
               taskGenerator.getMaxAttemptsPerTask()));
     }
     if (responseMap.isEmpty()) {
-      throw new NoTaskScheduledException("No task scheduled for 'tableName': " + tableName);
```

Comment:
Just curious why we throw exception here in the previous implementation, will both adhoc and scheduled  tasks handle this correctly?

Reply:
Not sure the reason for throwing an exception. In the schedule case, we log an info message. For schedule, its a valid case where there's no work to be done. However this can be the case with adhoc as well, depending on the underlying task implementation.

 LOGGER.info("No task to schedule for task type: {}", taskType);